timeboost-utils/src/lib.rs

```diff
@@ -61,3 +62,67 @@ pub fn select_peer_hosts(
         Box::new(keyset.iter().take(keyset.len())) as Box<dyn Iterator<Item = _>>
     }
 }
+
+/// Sometimes we don't want to reveal the struct inner fields when serializing.
+/// This intermediate type holds the bs58::encode(bincode::encode(T)) string that treats
+/// the type `T` as a `Serialize + Deserialize` blackbox.
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Blackbox(String);
```

Comment:
I don't fully understand the motivation here. Is `Blackbox` meant to be a type that (de-)serializes from/to bs58+bincode? Why does it hold a `String`?

Reply:
So if we serialize a struct in crypto, say EncryptionKey used for dkg, you will get:

```toml
[keys.dkg.pubilc]
u = "..."

[keys.dkg.secret]
alpha = "..."
```

revealing its internals. So we choose to serialize these T as `bs58::encode(bincode::encode(T)): String`, 
basically I use `Blackbox` to replace https://github.com/EspressoSystems/timeboost/blob/main/timeboost-utils/src/keyset.rs#L67-L95

Reply:
I added more rationale in [7d913c4](https://github.com/EspressoSystems/timeboost/pull/469/commits/7d913c43b82ea6d55f01dbfdc43924fd607c7bec), see if you find it reasonable. 

https://github.com/EspressoSystems/timeboost/blob/7d913c43b82ea6d55f01dbfdc43924fd607c7bec/timeboost-utils/src/lib.rs#L70-L93

Reply:
Why do `DkgEncKey` and `DkgDecKey` not implement `Serialize`/`Deserialize` directly in a way that is compact and readable, e.g. for human consumption with base 58?

The problem with `Blackbox` is that it erases all type information. Having this in a larger struct means one has to cast at runtime to some specific type. If some specific encoding should be used and serde's `with` attribute is not desired, why not make it a generic types, e.g. `Bs58Bincode<T>(T)`?

Reply:
adopted your code in [6a4edd4](https://github.com/EspressoSystems/timeboost/pull/469/commits/6a4edd4c64fcd7321f7567149f03c4414c29f1fd)